test/Conversion/MooreToCore/classes.mlir

```diff
@@ -80,3 +80,25 @@ moore.class.classdecl @E extends @C {
   moore.class.propertydecl @b : !moore.l32
   moore.class.propertydecl @c : !moore.l32
 }
+
+/// Check that upcast lowers to no-op
+
+// CHECK-LABEL: func.func private @test_new5
+// CHECK:   [[SIZE:%.*]] = llvm.mlir.constant(24 : i64) : i64
+// CHECK:   [[PTR:%.*]] = llvm.call @malloc([[SIZE]]) : (i64) -> !llvm.ptr
+// CHECK:   return
+
+// CHECK-NOT: moore.class.new
+// CHECK-NOT: moore.class.upcast
+// CHECK-NOT: moore.class.classdecl
+
+func.func private @test_new5() {
+  %h = moore.class.new : <@F>
+  %upcast = moore.class.upcast %h : <@F> to <@C>
+  return
+}
```

Comment:
How about passing class `%h` in as a func arg, such that the test doesn't cover `moore.class.new`, but only `moore.class.upcast`? Something like:
```mlir
// CHECK-LABEL: func.func private @test_new5
// CHECK-SAME: (%arg0: !llvm.ptr) -> !llvm.ptr
// CHECK-NEXT:   return %arg0 : !llvm.ptr

// CHECK-NOT: moore.class.upcast
// CHECK-NOT: moore.class.classdecl

func.func private @test_new5(%arg0: !moore.class<@F>) -> !moore.class<@C> {
  %0 = moore.class.upcast %arg0 : <@F> to <@C>
  return %0 : !moore.class<@C>
}
```

Reply:
Sure, done!